packages/mui-material/src/Chip/Chip.js

```diff
@@ -156,141 +146,197 @@ const ChipRoot = styled('div', {
+                  },
+                },
+              },
+            };
+          }),
+        {
+          props: (props) => props.iconColor === props.color,
+          style: {
+            [`& .${chipClasses.icon}`]: {
+              color: theme.vars ? theme.vars.palette.Chip.defaultIconColor : textColor,
+            },
           },
-        }),
+        },
+        {
+          props: (props) => props.iconColor === props.color && props.color !== 'default',
+          style: {
+            [`& .${chipClasses.icon}`]: {
+              color: 'inherit',
+            },
+          },
+        },
+        {
+          props: { onDelete: true },
+          style: {
+            [`&.${chipClasses.focusVisible}`]: {
+              backgroundColor: theme.vars
+                ? `rgba(${theme.vars.palette.action.selectedChannel} / calc(${theme.vars.palette.action.selectedOpacity} + ${theme.vars.palette.action.focusOpacity}))`
+                : alpha(
+                    theme.palette.action.selected,
+                    theme.palette.action.selectedOpacity + theme.palette.action.focusOpacity,
+                  ),
+            },
+          },
+        },
+        ...Object.entries(theme.palette)
+          .filter(([, value]) => value.dark)
+          .map(([color]) => {
+            return {
+              props: { color, onDelete: true },
+              style: {
+                [`&.${chipClasses.focusVisible}`]: {
+                  background: (theme.vars || theme).palette[color].dark,
+                },
+              },
+            };
+          }),
+      ],
     };
   },
-  ({ theme, ownerState }) => ({
-    ...(ownerState.clickable && {
-      userSelect: 'none',
-      WebkitTapHighlightColor: 'transparent',
-      cursor: 'pointer',
-      '&:hover': {
-        backgroundColor: theme.vars
-          ? `rgba(${theme.vars.palette.action.selectedChannel} / calc(${theme.vars.palette.action.selectedOpacity} + ${theme.vars.palette.action.hoverOpacity}))`
-          : alpha(
-              theme.palette.action.selected,
-              theme.palette.action.selectedOpacity + theme.palette.action.hoverOpacity,
-            ),
-      },
-      [`&.${chipClasses.focusVisible}`]: {
-        backgroundColor: theme.vars
-          ? `rgba(${theme.vars.palette.action.selectedChannel} / calc(${theme.vars.palette.action.selectedOpacity} + ${theme.vars.palette.action.focusOpacity}))`
-          : alpha(
-              theme.palette.action.selected,
-              theme.palette.action.selectedOpacity + theme.palette.action.focusOpacity,
-            ),
-      },
-      '&:active': {
-        boxShadow: (theme.vars || theme).shadows[1],
-      },
-    }),
-    ...(ownerState.clickable &&
-      ownerState.color !== 'default' && {
-        [`&:hover, &.${chipClasses.focusVisible}`]: {
-          backgroundColor: (theme.vars || theme).palette[ownerState.color].dark,
+  ({ theme }) => ({
+    variants: [
```

Comment:
With `variants`, there is no need to split the style args, all of them can be grouped into one `variants`.

Reply:
Fixed 🙌🏼